src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -258,28 +259,28 @@ private void SetInputPullMode(int pinNumber, PinMode mode)
              */
 
             uint* gppudPointer = &_registerViewPointer->GPPUD;
-            uint register = *gppudPointer;
-            register &= ~0b11U;
-            register |= modeToPullMode;
-            *gppudPointer = register;
+            *gppudPointer &= ~0b11U;
+            *gppudPointer |= modeToPullMode;
 
             // Wait 150 cycles – this provides the required set-up time for the control signal
-            Thread.SpinWait(150);
+            for (int i = 0; i < 150; i++)
+                ;
 
             int index = pinNumber / 32;
             int shift = pinNumber % 32;
             uint* gppudclkPointer = &_registerViewPointer->GPPUDCLK[index];
-            register = *gppudclkPointer;
-            register |= 1U << shift;
-            *gppudclkPointer = register;
+            uint pinBit = 1U << shift;
+            *gppudclkPointer |= pinBit;
 
             // Wait 150 cycles – this provides the required hold time for the control signal
-            Thread.SpinWait(150);
+            for (int i = 0; i < 150; i++)
+                ;
 
-            register = *gppudPointer;
-            register &= ~0b11U;
-            *gppudPointer = register;
-            *gppudclkPointer = 0;
+            *gppudclkPointer &= ~pinBit;
+            *gppudPointer &= ~0b11U;
+
+            for (int i = 0; i < 150; i++)
```

Comment:
Can you please update the comment above the process to point out that we had to go against spec because that was the way to make it work? That way if we ever have to change this we will take that into consideration.

Reply:
actually I'm not sure this wait is needed, I added this when testing - will double check...

Reply:
seems the delay might be needed - I'm getting 2/50 failures without it. Will double check my original results to make sure I wasn't just lucky with no failures

Reply:
100/100 passed with this timeout, 2/50 failed without it